src/plugins/explore/public/components/visualizations/bar/to_expression.ts

```diff
@@ -497,10 +513,14 @@ export const createStackedBarSpec = (
   const barMark: any = {
     type: 'bar',
     tooltip: styles.tooltipOptions?.mode !== 'hidden',
-    size: styles.barWidth ? styles.barWidth * 20 : 14, // Scale the bar width
-    binSpacing: styles.barPadding ? styles.barPadding * 10 : 1, // Scale the bar padding
   };
 
+  // Only set size and binSpacing in manual mode
+  if (styles.barSizeMode === 'manual') {
+    barMark.size = styles.barWidth ? styles.barWidth * 20 : 14;
+    barMark.binSpacing = styles.barPadding ? styles.barPadding * 10 : 1;
```

Comment:
Could you also extract this piece of code to a function? it's repeated many times.